### PR TITLE
glib/BindingBuilder: add concrete type based transform setters

### PR DESCRIPTION
**Pull request by fengalin**
Not sure about the naming though :)

Fixes https://github.com/gtk-rs/gtk-rs-core/issues/382

### CODE REVIEW

File: glib/src/object.rs [hunk 1]
```diff
@@ -3572,6 +3591,25 @@ impl<'a> BindingBuilder<'a> {
...
+    >(
+        self,
+        func: F,
+    ) -> Self {
+        Self {
+            transform_to: Some(Box::new(move |binding, from_value| {
+                let from_value = from_value.get().expect("Wrong value type");
+                func(binding, from_value).map(|r| r.to_value())
+            })),
+            ..self
```
Comment (GuillaumeGomez) [inline glib/src/object.rs:3609]: I didn't know it was a thing. :o

File: glib/src/object.rs [hunk 2]
```diff
@@ -3500,23 +3500,23 @@ impl<T: ObjectType> From<WeakRef<T>> for SendWeakRef<T> {
...
+pub struct BindingBuilder<'a, 'f, 't> {
     source: &'a ObjectRef,
     source_property: &'a str,
     target: &'a ObjectRef,
     target_property: &'a str,
     flags: crate::BindingFlags,
-    transform_to: TransformFn,
-    transform_from: TransformFn,
+    transform_from: TransformFn<'f>,
+    transform_to: TransformFn<'t>,
```
Comment (sdroege) [inline glib/src/object.rs:3516]: Why did you change the order? :)

File: glib/src/object.rs [hunk 3]
```diff
@@ -3560,6 +3560,25 @@ impl<'a> BindingBuilder<'a> {
         }
     }
 
+    // rustdoc-stripper-ignore-next
+    /// Transform changed property values from the target object to the source object with the given closure.
+    pub fn transform_from_concrete<
```
Comment (sdroege) [inline glib/src/object.rs:3565]: I would call this one `transform_from()` and the other one `transform_from_with_values()` or so

File: glib/src/object.rs (hunk 2 above)
Comment (fengalin) [reply glib/src/object.rs:3516]: So the signature for the Builder uses `<'a, 'f, 't>` instead of `<'a, 't, 'f>`. :)

File: glib/src/object.rs (hunk 3 above)
Comment (fengalin) [reply glib/src/object.rs:3565]: All right!

File: glib/src/object.rs (hunk 2 above)
Comment (sdroege) [reply glib/src/object.rs:3516]: OK, so no deeper reason :) Thanks

**Comment by fengalin** [comment]
Should be good now! :)

**Comment by sdroege** [comment]
@bilelmoussaoui What do you think?

**Review by bilelmoussaoui — APPROVED** [review]
